sdk/models.go

```diff
@@ -27,14 +28,22 @@ type Reading struct {
 
 // NewReading creates a new instance of a Reading. This is the recommended method
 // for creating new readings.
-func NewReading(output *Output, value interface{}) *Reading {
+// TODO: This needs to do a parameter check on output for sure since it's dereferencing a pointer.
+func NewReading(output *Output, value interface{}) (reading *Reading, err error) {
+	if output == nil {
+		return nil, fmt.Errorf("Unable to create reading. output is nil")
+	}
```

Comment:
@MatthewHink 

I have a bit of a philosophical/design question here. I think this check is valid, I'm just not sure whether it should result in an error being returned or if it should crash the plugin (e.g. either `panic` or `log.Fatal`) -- my thinking behind that is that the stuff where this is being used (e.g. the handlers) should know what outputs they support, and those outputs shouldn't be changing on the fly.

If the output is nil here (e.g. if `device.GetOutput("something")` returns nil -- no output found), then to me that means either:
- there is something wrong with the handler (typo in output name, e.g.)
- the plugin/device is misconfigured (it should support output X but does not)
- the outputs are changing dynamically - this is not supported and I don't think it should be.

In either of those cases, it seems like crashing the plugin seems right because this isn't something that can be recovered from, and if we just return or log an error, it will do so every time. I guess if someone is looking at the logs and recognizes this, then it is okay, but I'm not sure I'd want to bet on that.

Reply:
For strongly typed languages, crashing is not right. Imagine if the go compiler crashed and gave out a stack rather than a compiler error. Public interfaces need to validate parameter inputs before trying to use them.

Python is different (and strange) in this respect. It does whatever it can do with duck typing.

Reply:
I agree that this should validate the parameter and the check for nil should be there, I just also think that if the plugin is misconfigured to the point where it will not work (e.g. here, if the output is not specified, it will never be able to complete a read), then it makes sense to make a big deal out of it. 

I think I'd rather have it terminate with a `log.Fatal` saying that there is no such output and that there is a configuration error than have it continue on looking like its working until we try and read from it. Not sure if thats the right mindset for this condition, but thats where my head is at.

Reply:
Well then we're debugging crashing containers when the configuration isn't right. Kubernetes will attempt to restart it, and it will crash again.

It would be nice to have configuration validation, but configuration is constantly changing so it's hard to keep up with.

Reply:
fair enough!